src/ui/components/ReplayAssistButton.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { FC, useEffect, useState } from "react";
+import { useDispatch } from "react-redux";
+
+import { setSelectedPrimaryPanel } from "ui/actions/layout";
+import { toggleReplayAssist } from "ui/reducers/app";
+import useAuth0 from "ui/utils/useAuth0";
+
+const ReplayAssistButton: FC = () => {
+  const { isAuthenticated } = useAuth0();
+  const [shouldShowReplayAssist, setShouldShowReplayAssist] = useState(false);
+
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    const storedValue = localStorage.getItem("replayAssistEnabled");
+    if (storedValue) {
+      setShouldShowReplayAssist(JSON.parse(storedValue));
+    }
+  }, []);
+
+  useEffect(() => {
+    localStorage.setItem("replayAssistEnabled", JSON.stringify(shouldShowReplayAssist));
+  }, [shouldShowReplayAssist]);
+
+  if (!isAuthenticated) {
+    return null;
+  }
+
+  return (
+    <div className="row row-flat rounded-none border-t border-b">
+      <input
+        type="checkbox"
+        className="rounded-sm border-gray-400"
+        id="replay-assist-checkbox"
+        checked={shouldShowReplayAssist}
+        onChange={() => {
+          const newValue = !shouldShowReplayAssist;
+          setShouldShowReplayAssist(newValue);
+          dispatch(toggleReplayAssist());
```

Comment:
Mixing set-boolean and toggle actions runs the risk of mismatches. Replace `toggleReplayAssist` with a Redux setter to make sure they stay in sync:
```js
dispatch(setReplayAssist(newValue));
```

Reply:
Leaving a note for myself that I understand the point here, but not how to fix it.

Reply:
Hey @bvaughn I'm not super familiar with Redux setters and I'm having trouble getting this working. Can you share the steps I'll be taking so I can make sure I'm going in the right direction?

Reply:
I figured it out!